elki/src/main/java/elki/math/statistics/dependence/MCDEDependenceMeasure.java

```diff
@@ -0,0 +1,249 @@
+    final MCDETest.RankStruct[] index_1 = mcdeTest.corrected_ranks(adapter2, data2, len);
+
+    double mwp = 0;
+
+    for(int i = 0; i < this.m; i++) {
+      int r = random.nextInt(2);
+      MCDETest.RankStruct[] ref_index;
+      MCDETest.RankStruct[] other_index;
+
+      if(r == 1) {
+        ref_index = index_1;
+        other_index = index_0;
+      }
+      else {
+        ref_index = index_0;
+        other_index = index_1;
+      }
+
+      final int start = random.nextInt((int) (len * (1 - this.beta)));
+      final int end = start + (int) Math.ceil(len * this.beta);
+      mwp += mcdeTest.statistical_test(start,end, randomSlice(len, other_index), ref_index);
+    }
+    return mwp / m;
+  }
+
+  public static class Par implements Parameterizer {
+    /**
+     * Parameter that specifies the number of iterations in the Monte-Carlo
+     * process of identifying high contrast subspaces.
+     */
+    public static final OptionID M_ID = new OptionID("MCDE.test", "No. of Monte-Carlo iterations.");
+
+    /**
+     * Parameter that specifies the size of the slice
+     */
+    public static final OptionID ALPHA_ID = new OptionID("MCDE.test", "Expected share of instances in slice (independent dimensions).");
+
+    /**
+     * Parameter that specifies the size of the marginal restriction. Note that in the original paper
+     * alpha = beta and as such there is no explicit distinction between the parameters.
+     */
+    public static final OptionID BETA_ID = new OptionID("MCDE.test", "Expected share of instances in marginal restriction (dependent dimensions).");
+
+    /**
+     * Parameter that specifies the random seed.
+     */
+    public static final OptionID SEED_ID = new OptionID("MCDE.test", "The random seed.");
+
+    /**
+     * Parameter that specifies which mcde statistical test to use in order to
+     * calculate the contrast between the instances in the slice vs out of slice.
+     */
+    public static final OptionID TEST_ID = new OptionID("MCDE.test", "The mcde statistical test that is used to calculate the deviation of two data samples");
+
+
+    /**
+     * Holds the value of {@link #M_ID}.
+     */
+    protected int m = 50;
+
+    /**
+     * Holds the value of {@link #ALPHA_ID}.
+     */
+    protected double alpha = 0.5;
+
+    /**
+     * Holds the value of {@link #BETA_ID} .
+     */
+    protected double beta = 0.5;
+
+    /**
+     * Holds the value of {@link #TEST_ID}.
+     */
+    private MCDETest mcdeTest;
+
+    /**
+     * Random generator.
+     */
+    protected RandomFactory rnd;
+
+    @Override public void configure(Parameterization config) {
+      new IntParameter(M_ID, 50) //
+          .addConstraint(CommonConstraints.GREATER_THAN_ONE_INT).grab(config, x -> m = x);
+
+      new DoubleParameter(ALPHA_ID, 0.5) //
+          .addConstraint(CommonConstraints.GREATER_THAN_ZERO_DOUBLE).addConstraint(CommonConstraints.LESS_THAN_ONE_DOUBLE).grab(config, x -> alpha = x);
```

Comment:
these should be multiple lines (use ` //` to prevent the code formatter to join the lines).